tests/unit/test_download.py

```diff
@@ -713,6 +714,10 @@ def test_iter_secure_origins__trusted_hosts_empty(self):
             ("http://example.com/something/", ["example.com"], True),
             # Try changing the case.
             ("http://eXample.com/something/", ["example.cOm"], True),
+            # Test hosts with port.
+            ("http://example.com:8080/something/", ["example.com"], True),
+            ("http://example.com:8080/something/", ["example.com:8080"], True),
```

Comment:
Add a comment above this line saying, "Test a trusted_host with a port." And then you should probably also test a location with a port number where the number is different.